accepted/0000-basic-orchestration.md

```diff
@@ -0,0 +1,125 @@
+
+### Database and Hibernate changes
+
+In order to store the `jid` and the executions outcome a new table is needed:
+
+```sql
+CREATE TABLE IF NOT EXISTS rhnOrchestrationAction (
+
+    action_id       NUMERIC NOT NULL
+                            CONSTRAINT rhn_actsaltrunjob_aid_fk
+                            REFERENCES rhnAction (id)
+                            ON DELETE CASCADE,
+    jid              VARCHAR(100) NOT NULL,
+    status           NUMERIC NOT NULL
+                         CONSTRAINT rhn_server_action_status_fk
+                             REFERENCES rhnActionStatus (id),
+    result_code      NUMERIC,
+    result_msg       TEXT,
+    pickup_time      TIMESTAMPTZ,
+    completion_time  TIMESTAMPTZ,
+    created         TIMESTAMPTZ
+                        DEFAULT (current_timestamp) NOT NULL,
+    primary key (action_id, jid)
+);
+
+```
+
+A new collection will be added to the cluster actions (`ClusterJoinNodeAction`, `ClusterRemoveNodeAction`, etc):
+
+```java
+class ClusterJoinNodeAction {
+  [...]
+  private Set<ActionSaltRunnerJob> runnerJobs = new HashSet<>();
+  [...]
+}
+```
+
+The Hibernate mappings will need to be adapted:
+
+```xml
+<subclass name="com.redhat.rhn.domain.action.cluster.ClusterJoinNodeAction"
+        lazy="true" discriminator-value="516">
+    [...]
+    <set name="runnerJobs" outer-join="false" cascade="all" lazy="true"
+          inverse="true">
+        <key column="action_id"/>
+        <one-to-many
+                class="com.redhat.rhn.domain.action.ActionSaltRunnerJob" />
+    </set>
+    <join table="rhnActionClusterJoinNode">
+      [...]
+    </join>
+</subclass>
+
+```
+
+### Action result display
+
+Actions executed on a system are shown in the Events tab of the system details. A similar approach should be used for orchestration results. They should be displayed in the details of the target entity or in a generic orchestration UI.
+
+In the case of clusters, the UI should be enhanced to show a view similar to the system Events tab.
+
+In later versions, if more complex orchestration will be implemented that doesn't target a particular entity like clusters, a generic UI should be implemented to show orchestration results.
+
+### Cluster actions
+
+The existing actions targeting clusters will be changed to use orchestration where it makes sense:
+- `ClusterJoinNodeAction`
+- `ClusterRemoveNodeAction`
+- `ClusterUpgradeNodeAction`
+- `ClusterGroupRefreshNodesAction`
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future?
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Use the Thorium Complex Reactor instead of Orchestration ?
```

Comment:
What would be the consequences of using Thorium? 
Additionally, can we use original Salt Reactor and Thorium at the same time?